client/my-sites/plugins/plugins-browser/index.jsx

```diff
@@ -47,32 +38,30 @@ const PageViewTrackerWrapper = ( { category, selectedSiteId, trackPageViews } )
 	return null;
 };
 
-const PluginsBrowser = ( { trackPageViews = true, category, search, hideHeader } ) => {
+const PluginsBrowser = ( {
+	trackPageViews = true,
+	category,
+	search,
+	hideHeader,
+	pluginsResultsComponent,
```

Comment:
The unit tests are failing because this prop is required but [the tests](https://github.com/Automattic/wp-calypso/blob/cf8c35f3de37051a7f9b89cb766a24ec19762164/client/my-sites/plugins/plugins-browser/test/index.jsx#L131) haven't been updated to include it.

I don't think we want to do this anyway, why can't the browser figure out what results component to use based on the props? If there is a search, show search results, if there is a category, show the category?

Reply:
That sounds right, I tried to split this and make the logic in the route controller, but it makes things more complicated, I rollback these changes, and now all tests run fine.